### DISCUSSION TITLE
Basic SEO metatags first party plugin

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#10708] [Plugin ideas]
I think that some form of SEO functionality is used on 99% of all websites. 

Therefore, it would be logical if Craft had some first party SEO plugin. Seomatic is great, but in many cases it is just overkill and for simpliest/cheapest websites there is just no budget for it.

Such first party plugin should handle just simpliest things - title, description, keywords and open graph. It should have SEO field for element-specific settings and also global SEO settings.

**Comment by timkelty** [comment]
@piotrpog I think something more generic to support this idea, like reusable field sets/instances might be more appropriate for Craft. You can already achieve what you're describing with entries and globals, but the UI and DX of creating and reusing them isn't quite what you want.

**Reply by piotrpog** [reply]
Yes, i actually use Globals and fields along with great [seomate](https://github.com/vaersaagod/seomate) plugin. But thanks to using such method, various websites have varying seom fields setups. Different field names, different global names. Thats because seomate just generates metatags from fields and does not have any admin user interface.

It would be much better to have some unified solution, which some plugin providing admin UX would provide.

**Comment by piotrpog** [comment]
Hi,
could I get a statement from Craft devs, if some basic SEO functionality will be ever built in into craft or provided as the first party plugin? One of my clients asked about it.
@brandonkelly

**Reply by brandonkelly** [reply]
We’re currently focused on improving content modeling, which could improve the DX of manually setting up SEO fields, as @timkelty pointed out.

> Seomatic is great, but in many cases it is just overkill and for simpliest/cheapest websites there is just no budget for it.

There are a couple free, well-maintained alternatives to SEOmatic: [SEOMate by Værsågod ](https://plugins.craftcms.com/seomate?craft4) and [SEO by Ether](https://plugins.craftcms.com/seo?craft4)

**Reply by piotrpog** [reply]
@brandonkelly i actually use seomate a lot, but it requires fields setup and this can cause differences between various sites, sometimes developer adds specific field and sometimes not, field handles might be different and such.
